#!/usr/bin/env python
PACKAGE = "path_planner"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("goal_weight", int_t, 0, "the weight determining how much the goal will affect the potential of a point", 1, 0, 100)
gen.add("obstacle_weight", int_t, 0, "# the weight determining how much the obstacles will affect the potential of a point", 100, 0, 100)
gen.add("profile_diameter", int_t, 0, "n in the nxn square around the vessel for which potential is calculated", 50, 0, 100)
gen.add("d_inf", int_t, 0, "the euclidean distance inside which obstacles affect the potential of a point", 20, 0, 100)
gen.add("p_ngz", int_t, 0, "the potential added to a point inside the no-go zones", 50, 0, 100)
gen.add("p_hyst", int_t, 0, "the extra potential added to a point behind the vessel i.e the coast of tacking", 10, 0, 100)
gen.add("g_v", int_t, 0, "scalar multiplied to the velocity consideration", 10, 0, 100)
gen.add("circle_time_limit", int_t, 0, "The amount of time the circling maneuver is performed", 120, 0, 1000)
gen.add("wind_limit", double_t, 0, "The minimal wind velocity the path planner will notice knots", 1, 0, 10)

exit(gen.generate(PACKAGE, "path_planner", "PathPlanner"))
